### PR TITLE
ci: integrate automatic iam creation in e2e test

### PR DESCRIPTION
**Pull request by msanft**
### Proposed change(s)

- Add e2e testing of the `iam create` command. To run the test, use the `iamcreate` payload.

### Additional info
- [Successful GCP run](https://github.com/edgelesssys/constellation/actions/runs/4102278056)
- [Successful AWS run](https://github.com/edgelesssys/constellation/actions/runs/4124209934)
- The Azure MI / SP used for the CI does not have the permissions needed to update / assign / (...?) IAM policies, this will be adressed in a separate ticket [here](https://dev.azure.com/Edgeless/Edgeless/_backlogs/backlog/Edgeless%20Team/Features/?workitem=2912)

### Checklist

- [x] Add labels (e.g., for changelog category)
- [x] Link to Milestone

### CODE REVIEW

File: .github/actions/constellation_iam_create/action.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      shell: bash
+      if: inputs.cloudProvider == 'aws'
+      run: |
+        constellation iam create aws --zone=${{ inputs.awsZone }} --prefix=${{ inputs.awsPrefix }} --generate-config --yes
+
+    - name: Constellation iam create azure
+      shell: bash
+      if: inputs.cloudProvider == 'azure'
+      run: |
+        constellation iam create azure --region=${{ inputs.azureRegion }} --resourceGroup=${{ inputs.azureResourceGroup }} --servicePrincipal=${{ inputs.azureServicePrincipal }} --generate-config --yes
```
Comment (katexochen) [inline .github/actions/constellation_iam_create/action.yml:60]:
```suggestion
        constellation iam create azure \
            --region=${{ inputs.azureRegion }} \
            --resourceGroup=${{ inputs.azureResourceGroup }} \
            --servicePrincipal=${{ inputs.azureServicePrincipal }} \
            --generate-config \
            --yes
```
Break long line commands for better readability

File: .github/actions/constellation_iam_destroy/action.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+name: Delete IAM configuration
+description: Delete previously created IAM configuration.
+
+runs:
+  using: "composite"
+  steps:
+    - name: Delete IAM configuration
+      shell: bash
+      run: |
+        if [ -f constellation-iam-terraform/terraform.tfstate ]; then
```
Comment (katexochen) [inline .github/actions/constellation_iam_destroy/action.yml:10]:
```suggestion
        if [[ -f constellation-iam-terraform/terraform.tfstate ]]; then
```

**Review by katexochen — CHANGES_REQUESTED** [review]

**Comment by katexochen** [comment]
lgtm so far, could you rebase and trigger another run?

**Comment by msanft** [comment]
[Successful run after rebase](https://github.com/edgelesssys/constellation/actions/runs/4231457904/jobs/7350016752)

**Review by katexochen — APPROVED** [review]

**Pull request merged**